pkg/kubefed2/federate/federate.go

```diff
@@ -178,13 +192,20 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 		return nil, errors.Wrapf(err, "Error getting %s from %s %q", typeConfig.GetFederatedType().Kind, typeConfig.GetTarget().Kind, qualifiedName)
 	}
 
-	return &fedResources{
-		typeConfig:  typeConfig,
-		FedResource: fedResource,
-	}, nil
+	return fedResource, nil
 }
 
-func getTypeConfig(hostConfig *rest.Config, typeName, namespace string, outputYAML bool) (typeconfig.Interface, error) {
+func getTypeResources(hostConfig *rest.Config, typeName, namespace string, enableType, outputYAML bool) (*enable.TypeResources, error) {
+	if enableType {
```

Comment:
Why should `--enable-type` result in creating a federated resource with a generated type config even when a type config might already exist in the API?

Reply:
That was a mistake, updated!